src/env/tock/mod.rs

```diff
@@ -22,6 +21,52 @@ pub struct TockEnv {
     rng: TockRng256,
     store: Store<TockStorage>,
     upgrade_storage: Option<TockUpgradeStorage>,
+    io_channel: TockIOChannel,
+}
+
+pub struct TockIOChannel {}
+
+impl IOChannel for TockIOChannel {
+    fn recv_with_timeout(
+        &mut self,
+        buf: &mut [u8; 64],
+        timeout: isize,
+    ) -> Option<SendOrRecvStatus> {
+        let usb_interface = match usb_ctap_hid::recv_with_timeout(buf, Duration::from_ms(timeout)) {
+            Some(usb_ctap_hid::SendOrRecvStatus::Received(interface)) => interface,
+            Some(_) => {
+                return Some(SendOrRecvStatus::Error);
+            }
+            None => {
+                return None;
+            }
+        };
+        Some(SendOrRecvStatus::Received(transport_for_interface(
+            usb_interface,
+        )))
+    }
+    fn send_or_recv_with_timeout(
+        &mut self,
+        buf: &mut [u8; 64],
+        timeout: isize,
+        transport: Transport,
+    ) -> Option<SendOrRecvStatus> {
+        let interface = match transport {
```

Comment:
This is why I propose not to use transport in a proper API: We will want to extend it with NFC eventually, and then repurposing the data type for switching between HID channels might be awkward.